### ISSUE TITLE
Resolve vulnerability alerts

### ISSUE THREAD

**Issue by jercle** [a23au/awe-base-images#23]
Tracking resolution of the following alerts:

## Vulnerability CVE-2023-23914
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/50

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| LOW |	libcurl |	7.87.0-r2 | [CVE-2023-23914](https://avd.aquasec.com/nvd/cve-2023-23914) |

HSTS ignored on multiple requests

## Vulnerability CVE-2023-23915
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/51

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| LOW |	libcurl |	7.87.0-r2 | [CVE-2023-23915](https://avd.aquasec.com/nvd/cve-2023-23915) |

HSTS amnesia with --parallel

## Vulnerability CVE-2022-4203
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/42
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/34

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| MEDIUM |	libssl3 |	3.0.8-r0 |	CVE-2022-4203 |

X.509 Name Constraints Read Buffer Overflow

## Vulnerability CVE-2022-4304
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/43
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/35

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| MEDIUM |	libssl3	| 3.0.8-r0 |	CVE-2022-4304 |

A timing based side channel exists in the OpenSSL RSA Decryption implementation which could be sufficient to recover a plaintext across a network in a Bleichenbacher style attack. To achieve a successful decryption an attacker would have to be able to send a very large number of trial messages for decryption. The vulnerability affects all RSA padding modes: PKCS#1 v1.5, RSA-OEAP and RSASVE. For example, in a TLS connection, RSA is commonly used by a client to send an encrypted pre-master secret to the server. An attacker that had observed a genuine connection between a client and a server could use this flaw to send trial messages to the server and record the time taken to process them. After a sufficiently large number of messages the attacker could recover the pre-master secret used for the original connection and thus be able to decrypt the application data sent over that connection.

## Vulnerability CVE-2023-23916
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/49

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| MEDIUM |	libcurl |	7.87.0-r2 |	[CVE-2023-23916](https://avd.aquasec.com/nvd/cve-2023-23916)  |

HTTP multi-header compression denial of service

## Vulnerability CVE-2022-4450
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/44
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/36

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| HIGH |	libssl3	| 3.0.8-r0 |	CVE-2022-4450 |

The function PEM_read_bio_ex() reads a PEM file from a BIO and parses and decodes the "name" (e.g. "CERTIFICATE"), any header data and the payload data. If the function succeeds then the "name_out", "header" and "data" arguments are populated with pointers to buffers containing the relevant decoded data. The caller is responsible for freeing those buffers. It is possible to construct a PEM file that results in 0 bytes of payload data. In this case PEM_read_bio_ex() will return a failure code but will populate the header argument with a pointer to a buffer that has already been freed. If the caller also frees this buffer then a double free will occur. This will most likely lead to a crash. This could be exploited by an attacker who has the ability to supply malicious PEM files for parsing to achieve a denial of service attack. The functions PEM_read_bio() and PEM_read() are simple wrappers around PEM_read_bio_ex() and therefore these functions are also directly affected. These functions are also called indirectly by a number of other OpenSSL functions including PEM_X509_INFO_read_bio_ex() and SSL_CTX_use_serverinfo_file() which are also vulnerable. Some OpenSSL internal uses of these functions are not vulnerable because the caller does not free the header argument if PEM_read_bio_ex() returns a failure code. These locations include the PEM_read_bio_TYPE() functions as well as the decoders introduced in OpenSSL 3.0. The OpenSSL asn1parse command line application is also impacted by this issue.

## Vulnerability CVE-2022-25147
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/32
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/31
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/29

| Severity |	Package |	Fixed Version |	Link |
| ------------- | ------------- | ------------- | ------------- |
| CRITICAL |	apr-util-ldap	| 1.6.3-r0 |	CVE-2022-25147 |

Integer Overflow or Wraparound vulnerability in apr_base64 functions of Apache Portable Runtime Utility (APR-util) allows an attacker to write beyond bounds of a buffer. This issue affects Apache Portable Runtime Utility (APR-util) 1.6.1 and prior versions.

## Vulnerability CVE-2022-24963
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/28

| Severity |	Package |	Fixed Version |	Link |
| ------------- | ------------- | ------------- | ------------- |
| CRITICAL |	apr |	1.7.1-r0 |	CVE-2022-24963 |

Integer Overflow or Wraparound vulnerability in apr_encode functions of Apache Portable Runtime (APR) allows an attacker to write beyond bounds of a buffer. This issue affects Apache Portable Runtime (APR) version 1.7.0.

## Vulnerability CVE-2022-28331
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/30

| Severity |	Package |	Fixed Version |	Link |
| ------------- | ------------- | ------------- | ------------- |
| CRITICAL |	apr	| 1.7.1-r0	| CVE-2022-28331 |

On Windows, Apache Portable Runtime 1.7.0 and earlier may write beyond the end of a stack based buffer in apr_socket_sendv(). This is a result of integer overflow.

#### Vulnerability CVE-2023-0286
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/41
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/33

| Severity |	Package |	Fixed Version |	Link |
| ------------- | ------------- | ------------- | ------------- |
| CRITICAL |	libssl3	3.0.8-r0 |	CVE-2023-0286 |

There is a type confusion vulnerability relating to X.400 address processing inside an X.509 GeneralName. X.400 addresses were parsed as an ASN1_STRING but the public structure definition for GENERAL_NAME incorrectly specified the type of the x400Address field as ASN1_TYPE. This field is subsequently interpreted by the OpenSSL function GENERAL_NAME_cmp as an ASN1_TYPE rather than an ASN1_STRING. When CRL checking is enabled (i.e. the application sets the X509_V_FLAG_CRL_CHECK flag), this vulnerability may allow an attacker to pass arbitrary pointers to a memcmp call, enabling them to read memory contents or enact a denial of service. In most cases, the attack requires the attacker to provide both the certificate chain and CRL, neither of which need to have a valid signature. If the attacker only controls one of these inputs, the other input must already contain an X.400 address as a CRL distribution point, which is uncommon. As such, this vulnerability is most likely to only affect applications which have implemented their own functionality for retrieving CRLs over a network.

## Vulnerability CVE-2023-0401
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/48
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/40

| Severity |	Package |	Fixed Version |	Link |
| ------------- | ------------- | ------------- | ------------- |
| HIGH |	libssl3	| 3.0.8-r0 |	CVE-2023-0401 |

A NULL pointer can be dereferenced when signatures are being verified on PKCS7 signed or signedAndEnveloped data. In case the hash algorithm used for the signature is known to the OpenSSL library but the implementation of the hash algorithm is not available the digest initialization will fail. There is a missing check for the return value from the initialization function which later leads to invalid usage of the digest API most likely leading to a crash. The unavailability of an algorithm can be caused by using FIPS enabled configuration of providers or more commonly by not loading the legacy provider. PKCS7 data is processed by the SMIME library calls and also by the time stamp (TS) library calls. The TLS implementation in OpenSSL does not call these functions however third party applications would be affected if they call these functions to verify signatures on untrusted data.

## Vulnerability CVE-2023-0217
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/47
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/39

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| HIGH |	libssl3 |	3.0.8-r0	| CVE-2023-0217 |

An invalid pointer dereference on read can be triggered when an application tries to check a malformed DSA public key by the EVP_PKEY_public_check() function. This will most likely lead to an application crash. This function can be called on public keys supplied from untrusted sources which could allow an attacker to cause a denial of service attack. The TLS implementation in OpenSSL does not call this function but applications might call the function if there are additional security requirements imposed by standards such as FIPS 140-3.

## Vulnerability CVE-2023-0216
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/46
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/38

| Severity |	Package |	Fixed Version |	Link |
| ----- | ----- | ----- | ----- |
| HIGH |	libssl3	| 3.0.8-r0	| CVE-2023-0216 |

An invalid pointer dereference on read can be triggered when an application tries to load malformed PKCS7 data with the d2i_PKCS7(), d2i_PKCS7_bio() or d2i_PKCS7_fp() functions. The result of the dereference is an application crash which could lead to a denial of service attack. The TLS implementation in OpenSSL does not call this function however third party applications might call these functions on untrusted data.

## Vulnerability CVE-2023-0215
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/45
- [ ] https://github.com/a23au/awe-base-images/security/code-scanning/37

| Severity |	Package	| Fixed Version|	Link |
| ----- | ----- | ----- | ----- |
| HIGH |	libssl3	| 3.0.8-r0	| CVE-2023-0215 |

The public API function BIO_new_NDEF is a helper function used for streaming ASN.1 data via a BIO. It is primarily used internally to OpenSSL to support the SMIME, CMS and PKCS7 streaming capabilities, but may also be called directly by end user applications. The function receives a BIO from the caller, prepends a new BIO_f_asn1 filter BIO onto the front of it to form a BIO chain, and then returns the new head of the BIO chain to the caller. Under certain conditions, for example if a CMS recipient public key is invalid, the new filter BIO is freed and the function returns a NULL result indicating a failure. However, in this case, the BIO chain is not properly cleaned up and the BIO passed by the caller still retains internal pointers to the previously freed filter BIO. If the caller then goes on to call BIO_pop() on the BIO then a use-after-free will occur. This will most likely result in a crash. This scenario occurs directly in the internal function B64_write_ASN1() which may cause BIO_new_NDEF() to be called and will subsequently call BIO_pop() on the BIO. This internal function is in turn called by the public API functions PEM_write_bio_ASN1_stream, PEM_write_bio_CMS_stream, PEM_write_bio_PKCS7_stream, SMIME_write_ASN1, SMIME_write_CMS and SMIME_write_PKCS7. Other public API functions that may be impacted by this include i2d_ASN1_bio_stream, BIO_new_CMS, BIO_new_PKCS7, i2d_CMS_bio_stream and i2d_PKCS7_bio_stream. The OpenSSL cms and smime command line applications are similarly affected.

**Comment by jercle** [comment]
➜ trivy image jercle/awe-httpd-base
2023-02-22T13:34:09.017+1100    INFO    Vulnerability scanning is enabled
2023-02-22T13:34:09.017+1100    INFO    Secret scanning is enabled
2023-02-22T13:34:09.017+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T13:34:09.017+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T13:34:09.022+1100    INFO    Detected OS: alpine
2023-02-22T13:34:09.022+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T13:34:09.025+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
➜ trivy image jercle/awe-perl-base
2023-02-22T13:33:13.273+1100    INFO    Vulnerability scanning is enabled
2023-02-22T13:33:13.273+1100    INFO    Secret scanning is enabled
2023-02-22T13:33:13.273+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T13:33:13.273+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T13:33:14.961+1100    INFO    Detected OS: alpine
2023-02-22T13:33:14.961+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T13:33:14.964+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
➜ trivy image jercle/awe-perl-oracle
2023-02-22T13:33:22.996+1100    INFO    Vulnerability scanning is enabled
2023-02-22T13:33:22.996+1100    INFO    Secret scanning is enabled
2023-02-22T13:33:22.996+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T13:33:22.996+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T13:33:26.393+1100    INFO    Detected OS: alpine
2023-02-22T13:33:26.393+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T13:33:26.394+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-oracle (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
➜ trivy image jercle/awe-httpd-base
2023-02-22T13:34:09.017+1100    INFO    Vulnerability scanning is enabled
2023-02-22T13:34:09.017+1100    INFO    Secret scanning is enabled
2023-02-22T13:34:09.017+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T13:34:09.017+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T13:34:09.022+1100    INFO    Detected OS: alpine
2023-02-22T13:34:09.022+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T13:34:09.025+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
✘1 ➜ trivy image jercle/awe-httpd-base:1.0.11
2023-02-22T13:35:03.559+1100    INFO    Vulnerability scanning is enabled
2023-02-22T13:35:03.559+1100    INFO    Secret scanning is enabled
2023-02-22T13:35:03.559+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T13:35:03.559+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T13:35:03.564+1100    INFO    Detected OS: alpine
2023-02-22T13:35:03.564+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T13:35:03.565+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base:1.0.11 (alpine 3.17.1)

Total: 24 (UNKNOWN: 0, LOW: 2, MEDIUM: 5, HIGH: 10, CRITICAL: 7)

┌───────────────┬────────────────┬──────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│    Library    │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                            Title
                       │
├───────────────┼────────────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ apr           │ CVE-2022-24963 │ CRITICAL │ 1.7.0-r2          │ 1.7.1-r0      │ Integer Overflow or Wraparound vulnerability in apr_encode   │
│               │                │          │                   │               │ functions o ...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-24963                   │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-25147 │          │                   │               │ Integer Overflow or Wraparound vulnerability in apr_base64   │
│               │                │          │                   │               │ functions o ...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-25147                   │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-28331 │          │                   │               │ On Windows, Apache Portable Runtime 1.7.0 and earlier may    │
│               │                │          │                   │               │ write beyond the...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-28331                   │
├───────────────┼────────────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ apr-util      │ CVE-2022-25147 │          │ 1.6.1-r14         │ 1.6.3-r0      │ Integer Overflow or Wraparound vulnerability in apr_base64   │
│               │                │          │                   │               │ functions o ...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-25147                   │
├───────────────┤                │          │                   │               │
                       │
│ apr-util-ldap │                │          │                   │               │
                       │
│               │                │          │                   │               │
                       │
│               │                │          │                   │               │
                       │
├───────────────┼────────────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ libcrypto3    │ CVE-2023-0286  │          │ 3.0.7-r2          │ 3.0.8-r0      │ There is a type confusion vulnerability relating to X.400    │
│               │                │          │                   │               │ address proc ......
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0286                    │
│               ├────────────────┼──────────┤                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4450  │ HIGH     │                   │               │ The function PEM_read_bio_ex() reads a PEM file from a BIO   │
│               │                │          │                   │               │ and parses...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4450                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0215  │          │                   │               │ The public API function BIO_new_NDEF is a helper function    │
│               │                │          │                   │               │ used for str...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0215                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0216  │          │                   │               │ An invalid pointer dereference on read can be triggered when │
│               │                │          │                   │               │ an applic...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0216                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0217  │          │                   │               │ An invalid pointer dereference on read can be triggered when │
│               │                │          │                   │               │ an applic...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0217                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0401  │          │                   │               │ A NULL pointer can be dereferenced when signatures are being │
│               │                │          │                   │               │ verified ...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0401                    │
│               ├────────────────┼──────────┤                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4203  │ MEDIUM   │                   │               │ [openssl: X.509 Name Constraints Read Buffer Overflow]       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4203                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4304  │          │                   │               │ A timing based side channel exists in the OpenSSL RSA        │
│               │                │          │                   │               │ Decryption imple...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4304                    │
├───────────────┼────────────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ libcurl       │ CVE-2023-23916 │          │ 7.87.0-r1         │ 7.87.0-r2     │ [curl: HTTP multi-header compression denial of service]      │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-23916                   │
│               ├────────────────┼──────────┤                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-23914 │ LOW      │                   │               │ [curl: HSTS ignored on multiple requests]                    │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-23914                   │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-23915 │          │                   │               │ [curl: HSTS amnesia with --parallel]
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-23915                   │
├───────────────┼────────────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ libssl3       │ CVE-2023-0286  │ CRITICAL │ 3.0.7-r2          │ 3.0.8-r0      │ There is a type confusion vulnerability relating to X.400    │
│               │                │          │                   │               │ address proc ......
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0286                    │
│               ├────────────────┼──────────┤                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4450  │ HIGH     │                   │               │ The function PEM_read_bio_ex() reads a PEM file from a BIO   │
│               │                │          │                   │               │ and parses...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4450                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0215  │          │                   │               │ The public API function BIO_new_NDEF is a helper function    │
│               │                │          │                   │               │ used for str...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0215                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0216  │          │                   │               │ An invalid pointer dereference on read can be triggered when │
│               │                │          │                   │               │ an applic...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0216                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0217  │          │                   │               │ An invalid pointer dereference on read can be triggered when │
│               │                │          │                   │               │ an applic...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0217                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2023-0401  │          │                   │               │ A NULL pointer can be dereferenced when signatures are being │
│               │                │          │                   │               │ verified ...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-0401                    │
│               ├────────────────┼──────────┤                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4203  │ MEDIUM   │                   │               │ [openssl: X.509 Name Constraints Read Buffer Overflow]       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4203                    │
│               ├────────────────┤          │                   │               ├──────────────────────────────────────────────────────────────┤
│               │ CVE-2022-4304  │          │                   │               │ A timing based side channel exists in the OpenSSL RSA        │
│               │                │          │                   │               │ Decryption imple...
                       │
│               │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-4304                    │
└───────────────┴────────────────┴──────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

**Comment by jercle** [comment]
```bash
➜ trivy image jercle/awe-httpd-base:1.0.11
2023-02-22T14:48:37.962+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:48:37.962+1100    INFO    Secret scanning is enabled
2023-02-22T14:48:37.962+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:48:37.962+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:48:40.904+1100    INFO    Detected OS: alpine
2023-02-22T14:48:40.904+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:48:40.905+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base:1.0.11 (alpine 3.17.1)

Total: 24 (UNKNOWN: 0, LOW: 2, MEDIUM: 5, HIGH: 10, CRITICAL: 7)
```
```bash
➜ trivy image jercle/awe-httpd-base
2023-02-22T14:49:52.038+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:49:52.038+1100    INFO    Secret scanning is enabled
2023-02-22T14:49:52.038+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:49:52.038+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:49:52.042+1100    INFO    Detected OS: alpine
2023-02-22T14:49:52.042+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:49:52.044+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by jercle** [comment]
➜ trivy image jercle/awe-httpd-base
2023-02-22T14:49:52.038+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:49:52.038+1100    INFO    Secret scanning is enabled
2023-02-22T14:49:52.038+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:49:52.038+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:49:52.042+1100    INFO    Detected OS: alpine
2023-02-22T14:49:52.042+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:49:52.044+1100    INFO    Number of language-specific files: 0

jercle/awe-httpd-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
```bash
➜ trivy image jercle/awe-perl-base:1.0.11
2023-02-22T14:50:01.915+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:50:01.915+1100    INFO    Secret scanning is enabled
2023-02-22T14:50:01.915+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:50:01.915+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:50:04.495+1100    INFO    Detected OS: alpine
2023-02-22T14:50:04.495+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:50:04.498+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-base:1.0.11 (alpine 3.17.1)

Total: 24 (UNKNOWN: 0, LOW: 2, MEDIUM: 5, HIGH: 10, CRITICAL: 7)
```
```bash
➜ trivy image jercle/awe-perl-base
2023-02-22T14:50:17.755+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:50:17.755+1100    INFO    Secret scanning is enabled
2023-02-22T14:50:17.755+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:50:17.755+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:50:17.760+1100    INFO    Detected OS: alpine
2023-02-22T14:50:17.760+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:50:17.763+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by jercle** [comment]
➜ trivy image jercle/awe-perl-base
2023-02-22T14:50:17.755+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:50:17.755+1100    INFO    Secret scanning is enabled
2023-02-22T14:50:17.755+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:50:17.755+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:50:17.760+1100    INFO    Detected OS: alpine
2023-02-22T14:50:17.760+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:50:17.763+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-base (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

**Comment by jercle** [comment]
```bash
➜ trivy image jercle/awe-perl-oracle:1.0.11
2023-02-22T14:54:48.377+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:54:48.377+1100    INFO    Secret scanning is enabled
2023-02-22T14:54:48.377+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:54:48.377+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:54:51.364+1100    INFO    Detected OS: alpine
2023-02-22T14:54:51.365+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:54:51.368+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-oracle:1.0.11 (alpine 3.17.1)

Total: 24 (UNKNOWN: 0, LOW: 2, MEDIUM: 5, HIGH: 10, CRITICAL: 7)
```
```bash
➜ trivy image jercle/awe-perl-oracle
2023-02-22T14:55:05.478+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:55:05.478+1100    INFO    Secret scanning is enabled
2023-02-22T14:55:05.478+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:55:05.478+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:55:05.483+1100    INFO    Detected OS: alpine
2023-02-22T14:55:05.483+1100    INFO    Detecting Alpine vulnerabilities...
2023-02-22T14:55:05.487+1100    INFO    Number of language-specific files: 0

jercle/awe-perl-oracle (alpine 3.17.1)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by jercle** [comment]
```bash
➜ trivy image jercle/awe-tomcat-jdk7-corretto:1.0.11
2023-02-22T14:55:22.839+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:55:22.839+1100    INFO    Secret scanning is enabled
2023-02-22T14:55:22.839+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:55:22.839+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:55:25.801+1100    INFO    Detected OS: amazon
2023-02-22T14:55:25.801+1100    INFO    Detecting Amazon Linux vulnerabilities...
2023-02-22T14:55:25.806+1100    INFO    Number of language-specific files: 1
2023-02-22T14:55:25.806+1100    INFO    Detecting jar vulnerabilities...

jercle/awe-tomcat-jdk7-corretto:1.0.11 (amazon 2 (Karoo))

Total: 26 (UNKNOWN: 0, LOW: 6, MEDIUM: 6, HIGH: 11, CRITICAL: 3)
```
```bash
➜ trivy image jercle/awe-tomcat-jdk7-corretto
2023-02-22T14:55:38.171+1100    INFO    Vulnerability scanning is enabled
2023-02-22T14:55:38.171+1100    INFO    Secret scanning is enabled
2023-02-22T14:55:38.172+1100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2023-02-22T14:55:38.172+1100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.36/docs/secret/scanning/#recommendation for faster secret detection
2023-02-22T14:55:38.180+1100    INFO    Detected OS: amazon
2023-02-22T14:55:38.180+1100    INFO    Detecting Amazon Linux vulnerabilities...
2023-02-22T14:55:38.183+1100    INFO    Number of language-specific files: 1
2023-02-22T14:55:38.183+1100    INFO    Detecting jar vulnerabilities...

jercle/awe-tomcat-jdk7-corretto (amazon 2 (Karoo))

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by jercle** [comment]
Closed with v1.0.12
